src/cartogram_info/read_csv.cpp

```diff
@@ -70,39 +70,104 @@ void CartogramInfo::read_csv(const argparse::ArgumentParser &arguments)
     // Get target area
     csv::CSVField area_field = row[area_col];
     double area;
+    std::string area_as_str = area_field.get();
+    char comma = ',';
+    char decimal = '.';
+    long unsigned int string_size = area_as_str.length();
     if (area_field.is_num()) {
       area = area_field.get<double>();
       if (area < 0.0) {
         std::cerr << "ERROR: negative area in CSV" << std::endl;
         _Exit(101);
       }
-    } else {
-      std::string area_as_str = area_field.get();
+    }
+
+    // Check for both commas and decimals
+    else if (
+      (area_as_str.find(comma) != std::string::npos) &&
+      (area_as_str.find(decimal) != std::string::npos)) {
+
+      // if commas come first, remove all commas
+      if (area_as_str.find(comma) < area_as_str.find(decimal)) {
+        area_as_str.erase(
+          std::remove(area_as_str.begin(), area_as_str.end(), comma),
+          area_as_str.end());
+        area = std::stod(area_as_str);
+      }
+
+      // Otherwise, remove decimals, then convert comma to decimal
+      else {
+        area_as_str.erase(
+          std::remove(area_as_str.begin(), area_as_str.end(), decimal),
+          area_as_str.end());
+        area_as_str[area_as_str.find(comma)] = '.';
+        area = std::stod(area_as_str);
+      }
+    }
+
+    // Check for commas only
+    else if (
+      (area_as_str.find(comma) != std::string::npos) &&
+      (area_as_str.find(decimal) == std::string::npos)) {
+      int count_commas = 0;
+      long unsigned int j;
+      for (j = 0; j < string_size; j++)
+        if ((area_as_str[j] == comma))
+          count_commas++;
+
+      // if there is only 1 comma, and there are only 2 numbers after the
+      // comma, convert to decimal
+      if (
+        (count_commas = 1) &&
+        ((area_as_str.find(comma) + 3) > (string_size))) {
+        area_as_str[area_as_str.find(comma)] = '.';
+      }
 
       // With inspiration from:
       // https://stackoverflow.com/questions/2684491/remove-commas-from-string
-      area_as_str.erase(
-        std::remove(area_as_str.begin(), area_as_str.end(), ','),
-        area_as_str.end());
+      // Otherwise, remove all commas
+      else {
+        area_as_str.erase(
+          std::remove(area_as_str.begin(), area_as_str.end(), comma),
+          area_as_str.end());
+      }
+      if (std::all_of(area_as_str.begin(), area_as_str.end(), ::isdigit)) {
```

Comment:
There is another issue. This line 134 condition will yield false if there are any decimals or commas in the string. In line 120 condition, if true, we insert a decimal (suppose 182,52 become 182.52). This will make the condition in line 134 false.

Reply:
From what I've seen, our code actually handles decimals fine. Values of .00, or even those with values appear to be reflected accurately. I might be wrong, so if that is the case please do raise this issue to me again!